tests/contenttypes_tests/test_operations.py

```diff
@@ -1,66 +1,129 @@
-        self.assertTrue(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedfoo').exists())
+        self.assertFalse(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedfoo').exists())
+        self.assertTrue(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedtwicefoo').exists())
         call_command('migrate', 'contenttypes_tests', 'zero', database='default', interactive=False, verbosity=0)
         self.assertTrue(ContentType.objects.filter(app_label='contenttypes_tests', model='foo').exists())
+        self.assertFalse(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedtwicefoo').exists())
         self.assertFalse(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedfoo').exists())
 
     def test_content_type_rename_conflict(self):
         ContentType.objects.create(app_label='contenttypes_tests', model='foo')
         ContentType.objects.create(app_label='contenttypes_tests', model='renamedfoo')
+        ContentType.objects.create(app_label='contenttypes_tests', model='renamedtwicefoo')
         call_command('migrate', 'contenttypes_tests', database='default', interactive=False, verbosity=0)
         self.assertTrue(ContentType.objects.filter(app_label='contenttypes_tests', model='foo').exists())
         self.assertTrue(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedfoo').exists())
+        self.assertTrue(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedtwicefoo').exists())
         call_command('migrate', 'contenttypes_tests', 'zero', database='default', interactive=False, verbosity=0)
         self.assertTrue(ContentType.objects.filter(app_label='contenttypes_tests', model='foo').exists())
         self.assertTrue(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedfoo').exists())
+        self.assertTrue(ContentType.objects.filter(app_label='contenttypes_tests', model='renamedtwicefoo').exists())
+
+
+@override_settings(
+    MIGRATION_MODULES=dict(
+        contenttypes_tests='contenttypes_tests.noop_migrations',
+    ),
+)
+class ContentTypeNotInstalledOperationsTests(TransactionTestCase):
+    available_apps = [
+        'contenttypes_tests',
+    ]
+
+    def setUp(self):
+        models.signals.post_operation.connect(
+            self.assertOperationsNotInjected,
+            sender=contenttypes_management.RenameContentType,
+        )
+        self.addCleanup(
+            models.signals.post_operation.disconnect,
+            self.assertOperationsNotInjected,
+            sender=contenttypes_management.RenameContentType,
+        )
+        self.addCleanup(
+            call_command,
+            'migrate',
+            'contenttypes_tests',
+            'zero',
+            database='default',
+            interactive=False,
+            verbosity=0,
+        )
```

Comment:
I guess this is required to make `--keepdb` work? Calling `call_command` sounds like the best option here.